File: scheduler-core/src/main/java/com/averude/uksatse/scheduler/core/dto/BasicDto.java
package com.averude.uksatse.scheduler.core.dto;

import lombok.AllArgsConstructor;
import lombok.Data;

import java.io.Serializable;
import java.util.List;

@Data
@AllArgsConstructor
public class BasicDto<P, C> implements Serializable {
    private P parent;
    private List<C> collection;
}

File: frontend-server/src/app/model/dto/basic-dto.ts
import { IdEntity } from "../interface/id-entity";

export class BasicDTO<E extends IdEntity, C> implements IdEntity {
  parent:     E;
  collection: C[];

  get id(): number {
    return this.parent.id;
  }

  set id(id: number) {
    this.parent.id = id;
  }
}
